packages/client/src/runtime/utils/getClientEngineType.ts

```diff
@@ -0,0 +1,35 @@
+import { GeneratorConfig } from '@prisma/generator-helper'
+
+export enum ClientEngineType {
+  NodeAPI = 'node-api',
+  Binary = 'binary',
+}
+export const DEFAULT_CLIENT_ENGINE_TYPE = ClientEngineType.Binary
+
+export function getClientEngineType(
+  generator?: GeneratorConfig,
+): ClientEngineType {
+  const engineTypeFromEnvVar = getEngineTypeFromEnvVar()
+  if (engineTypeFromEnvVar) return engineTypeFromEnvVar
+  if (
+    generator?.config.engineType === ClientEngineType.NodeAPI ||
+    generator?.previewFeatures.includes('nApi')
+  ) {
+    return ClientEngineType.NodeAPI
+  } else if (generator?.config.engineType === ClientEngineType.Binary) {
+    return ClientEngineType.Binary
+  }
+  return DEFAULT_CLIENT_ENGINE_TYPE
+}
+
+function getEngineTypeFromEnvVar() {
+  const engineType = process.env.PRISMA_CLIENT_ENGINE_TYPE
+  if (engineType) {
+    if (engineType === ClientEngineType.NodeAPI) {
+      return ClientEngineType.NodeAPI
+    } else if (engineType === ClientEngineType.Binary) {
+      return ClientEngineType.Binary
+    }
+  }
+  return undefined
```

Comment:
Could be "simplified" with

```suggestion
if (engineType === ClientEngineType.NodeAPI) {
  return ClientEngineType.NodeAPI
} else if (engineType === ClientEngineType.Binary) {
  return ClientEngineType.Binary
} else {
  return undefined
}
```